PyInstaller/__main__.py

```diff
@@ -102,22 +103,20 @@ def run(pyi_args=None, pyi_config=None):
     check_requirements()
 
     import PyInstaller.log
+    from PyInstaller.compat import is_conda
```

Comment:
> Why did you shift the import here? Top level imports should be fine.
- @Legorooj  ([original comment](
https://github.com/pyinstaller/pyinstaller/pull/6125#discussion_r689269074))
> Because the comment at the top of the file states we should not be importing anything else until check_requirements is run...
- @rokm ([original comment](https://github.com/pyinstaller/pyinstaller/pull/6125#discussion_r689306262))

Reply:
Yes, however `is_conda` is imported from the same module as `check_requirements` so splitting it up is pointless.

Reply:
Ah, right, of course. 

Mind if we do `from PyInstaller import compat`, then? And use `compat.check_requirements` and `compat.is_conda`?

Reply:
That would be fine, yeah.